hollow/src/main/java/com/netflix/hollow/core/read/engine/object/HollowObjectTypeReadStateShard.java

```diff
@@ -34,286 +31,127 @@
 import java.util.List;
 
 class HollowObjectTypeReadStateShard {
-
-    private volatile HollowObjectTypeDataElements currentDataVolatile;
+    final HollowObjectTypeDataElements dataElements;
+    final int shardOrdinalShift;
 
     private final HollowObjectSchema schema;
-    
-    HollowObjectTypeReadStateShard(HollowObjectSchema schema) {
+
+    HollowObjectTypeReadStateShard(HollowObjectSchema schema, HollowObjectTypeDataElements dataElements, int shardOrdinalShift) {
         this.schema = schema;
+        this.shardOrdinalShift = shardOrdinalShift;
+        this.dataElements = dataElements;
     }
 
-    public boolean isNull(int ordinal, int fieldIndex) {
-        HollowObjectTypeDataElements currentData;
-        long fixedLengthValue;
-
-        do {
-            currentData = this.currentDataVolatile;
-
-            long bitOffset = fieldOffset(currentData, ordinal, fieldIndex);
-            int numBitsForField = currentData.bitsPerField[fieldIndex];
-
-            fixedLengthValue = numBitsForField <= 56 ?
-                    currentData.fixedLengthData.getElementValue(bitOffset, numBitsForField)
-                    : currentData.fixedLengthData.getLargeElementValue(bitOffset, numBitsForField);
-        } while(readWasUnsafe(currentData));
-
-        switch(schema.getFieldType(fieldIndex)) {
-        case BYTES:
-        case STRING:
-            int numBits = currentData.bitsPerField[fieldIndex];
-            return (fixedLengthValue & (1L << (numBits - 1))) != 0;
-        case FLOAT:
-            return (int)fixedLengthValue == HollowObjectWriteRecord.NULL_FLOAT_BITS;
-        case DOUBLE:
-            return fixedLengthValue == HollowObjectWriteRecord.NULL_DOUBLE_BITS;
-        default:
-            return fixedLengthValue == currentData.nullValueForField[fieldIndex];
-        }
+    public long isNull(int ordinal, int fieldIndex) {
```

Comment:
Feels like this method has the wrong name now -- it is now just reading the actual field data whether or not it is beyond the max bits for a single (possibly unaligned) read.

Reply:
Good point, renamed to `readValue`